src/Components/Footer.js

```diff
@@ -1,58 +1,325 @@
-                            <span className="fa fa-github fa-lg"></span>
-                        </a>
-                        <a href="#" target="_blank" className="text-info">
-                            <span className="fa fa-linkedin fa-lg"></span>
-                        </a>
-                        <a href="#" target="_blank" className="text-primary">
-                            <span className="fa fa-twitter fa-lg"></span>
-                        </a>
-                        <a href="#" target="_blank" className="text-danger">
-                            <span className="fa fa-instagram fa-lg"></span>
-                        </a>
-                    </div>                                   
-                    {authState.auth.PK?
-                        <div className="col-4 col-sm-2 d-flex justify-content-around">
-                            <span onClick={toggleLightMode}>
-                                {!isDarkMode?
-                                    <span className="fa fa-moon-o fa-lg"></span>
-                                :
-                                    <span className="fa fa-sun-o fa-lg"></span>
-                                }
-                            </span>
-                            <NavLink to='/profile'><span className="fa fa-cog fa-lg"></span></NavLink>
-                            <NavLink to='/'><span className="fa fa-sign-out fa-lg" onClick={handleLogout}></span></NavLink>                                                                                    
-                        </div>     
-                    :   <div></div>
-                    }
-                    <div className="col-12 col-sm-4">
-                        v-transfer-demo@gmail.com
-                    </div>                
-                </div>
+    setError(errors);
+    setIsValid(isValid);
+  };
+
+  const updatePassword = (e) => {
+    e.preventDefault();
+    const oldPassword = e.target[0].value;
+    const newPassword = e.target[1].value;
+    if (isValid) {
+      Auth.currentAuthenticatedUser()
+        .then((user) => {
+          return Auth.changePassword(user, oldPassword, newPassword);
+        })
+        .then((data) => {
+          toast.success("Password updated successfully");
+          toggleMain();
+        })
+        .catch((err) => {
+          if (err.message === "Incorrect username or password.")
+            toast.error("Incorrect password");
+          else toast.error(err.message);
+        });
+    }
+  };
+
+  const updateUsername = async (e) => {
+    e.preventDefault();
+    const user = await Auth.currentAuthenticatedUser();
+    const newName = e.target[0].value;
+    Auth.updateUserAttributes(user, {
+      name: newName,
+    })
+      .then((res) => {
+        toast.success("User name updated successfully");
+        toggleMain();
+      })
+      .catch((err) => toast.error(err.message));
+  };
```

Comment:
Instead of requesting amplify to update the username first check locally is username empty or not than request.